sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -172,7 +173,7 @@ class V2SessionCatalog(sessionState: SessionState) extends TableCatalog {
 /**
  * An implementation of catalog v2 [[Table]] to expose v1 table metadata.
  */
-case class CatalogTableAsV2(v1Table: CatalogTable) extends Table {
+case class CatalogTableAsV2(v1Table: CatalogTable) extends UnresolvedTable {
```

Comment:
If we move `CatalogTableAsV2` to catalyst and rename it to `UnresolvedTable`, then we don't need to create an extra interface. what do you think? @brkyvz @rdblue

Reply:
`CatalogTable` is defined in sql unfortunately.

Reply:
It's defined in catalyst: `org.apache.spark.sql.catalyst.catalog.CatalogTable` in file `org/apache/spark/sql/catalyst/catalog/interface.scala`

Reply:
Oh, I like that a lot more